packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -160,12 +161,23 @@ abstract contract ForwarderModule is BaseModule
         // wallet funds.
         if (metaTx.gasPrice > 0 && (metaTx.txAwareHash == 0 || success)) {
             uint gasAmount = gasUsed < metaTx.gasLimit ? gasUsed : metaTx.gasLimit;
+
+            // Do not consume quota when call factory's createWallet function or
+            // when a successful meta-tx's txAwareHash is non-zero (which means it will
+            // be signed by at least a guardian). Therefor, even if the owner's
+            // private key is leaked, the hacker won't be able to deplete ether/tokens
+            // as high meta-tx fees.
+            bool skipQuota = metaTx.txAwareHash != 0 && success ||
+                metaTx.data.toBytes4(0) == WalletFactory.createWallet.selector &&
+                metaTx.to == controller().walletFactory();
```

Comment:
Will need to have an extra `&& success` for the `createWallet` case as well, we can't let it fail because that would allow it to be called an infinite number of times.

Reply:
Good point.

Reply:
fixed